cirq/api/google/v2/metrics.proto

```diff
@@ -0,0 +1,79 @@
+syntax = "proto3";
+
+package cirq.api.google.v1;
+
+option java_package = "com.google.cirq.api.google.v2";
+option java_outer_classname = "CalibrationProto";
+option java_multiple_files = true;
+
+// Performance metrics for a quantum processor.
+message Metrics {
+
+  // The time the metrics were collected, in milliseconds since Jan 1, 1970.
+  // Metric collection take time, so this is the time at which all of the
+  // metrics have been collected.
+  uint64 collection_time_ms = 1;
+
+  // Map from a metric type to a collection of the metrics all of this type.
+  // Keys should be snake_case and fairly descriptive of the metric.
+  map<string, MetricCollection> metrics = 2;
+}
+
+// A collection of metrics of the same type.
+message MetricCollection {
+  // The metrics in the collection. These should all be of the same type.
```

Comment:
Why not have the oneof at this level if we're going to require that they all be the same type?

Reply:
you can't have repeated oneofs, so if you do that you will end up making container classes for each metric type.  Do you think that's better?

Reply:
I think we could encapsulate this into the "target" of a metric, with a special all target. It will also give us flexibility to talk about subsets of qubits.